### PR TITLE
Diagram schema migration

### PR DESCRIPTION
**Pull request by loreanvictor**
### Checklist
- [x] I documented the TypeScript code using JSDoc style.
- ~~I added multiple screenshots/screencasts of my UI changes~~ (NA)
- ~~I translated all the newly inserted strings into German and English~~ (NA)

### Motivation and Context

Following up on [this discussion](https://github.com/ls1intum/Apollon/pull/297): Current model schema of Apollon results in unnecessary conflicts when patching / merging two replicas (e.g. in realtime collaboration) using [JSONPatch](https://jsonpatch.com) and [JSONPointer](https://gregsdennis.github.io/Manatee.Json/usage/pointer.html) standards.

### Description

The proposed change will use `id -> element` maps instead of arrays for `element`, `relationships`, `assessments` and `interactive` fields of the model schema, causing standard patch operations on these collections to become idempotent, and if not directly targeting the same element, commutative, effectively resolving most conflicts that could arise while merging two replicas of the model. It results in a major bump in model version to v3, with import support for v2 models (no export support added atm).

> **IMPORTANT**
>
> This requires a follow up PR for simplifying and optimising the [serialisation/deserialisation code](https://github.com/ls1intum/Apollon/blob/develop/src/main/components/store/model-state.ts). This change is excluded from the current PR as the existing code [has issues](https://github.com/ls1intum/Apollon/issues/312) that need to be fixed beforehand (as otherwise, it is almost impossible to ensure the refactoring yields correct results).

### Steps for Testing

1. Run the editor
2. Create a diagram and export it as JSON

### Test Coverage

| File | Branch | Line |
|------|-------:|-----:|
|main/apollon-editor.ts | 79.16% | 83.47% |
|main/typings.ts | 100% | 100% |
|main/compat | 100% | 100% |
|main/compat/v2 | 100% | 100% |
|main/components/store | 97.72% | 93.82% |
|main/packages/uml-communication-diagram/uml-communication-link/uml-communication-link.ts | 72.72% | 90.47% |
|main/scenes/svg.tsx | 73.46% | 81.56% |

There are some other files that are changed by prettier, I have not included them since their coverage is unchanged.

### CODE REVIEW

File: src/main/apollon-editor.ts [hunk 1]
```diff
@@ -372,6 +372,8 @@ export class ApollonEditor {
         this.store.getState().lastAction.endsWith('DELETE')
       ) {
         const lastModel = ModelState.toModel(this.store.getState());
+        // eslint-disable-next-line no-console
+        console.log(lastModel);
```
Comment (matthiaslehnertum) [inline src/main/apollon-editor.ts:376]: Can this be removed or is this log statement required for some sort of debugging?

Comment (loreanvictor) [reply src/main/apollon-editor.ts:376]: shoot that's for debugging, i'll remove it from the final code.

**Review by matthiaslehnertum — APPROVED** [review]

**Comment by matthiaslehnertum** [comment]
Looks good to me! Nicely done! Apollon itself as well as the deserialization of v2 diagrams operate as expected.

**Comment by matthiaslehnertum** [comment]
Can you merge this one @loreanvictor ? :)

**Comment by loreanvictor** [comment]
> Can you merge this one @loreanvictor ? :)

Unfortunately, I do not seem to have the permission 😅

**Comment by krusche** [comment]
I will merge the PR

**Pull request merged**